shared/chat/inbox/row/add-new-row.js

```diff
@@ -10,87 +10,110 @@ if (!isMobile) {
-      onClick={onNewChat}
-    >
-      <Icon type="iconfont-new" style={{color: globalColors.blue, marginRight: 9}} />
-      <Text type="BodyBigLink">New chat</Text>
-    </ClickableBox>
-    {isLoading &&
-      <Box style={{bottom: 0, left: 0, position: 'absolute', right: 0}}>
-        <LoadingLine />
-      </Box>}
-  </Box>
-)
+}
+
+type State = {
+  inEditingMode: boolean,
+  filter: string,
+}
+
+class _AddNewRow extends Component<void, Props, State> {
+  state: State
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      inEditingMode: false,
+      filter: '',
+    }
+  }
+
+  _setState = (inEditingMode, filter) => {
+    this.setState({inEditingMode, filter})
+    this.props.onSetFilter(filter)
+  }
 
-// const _AddNewRow= ({
-// onNewChat,
-// onJumpToChat,
-// isLoading,
-// }: {
-// onNewChat: () => void,
-// onJumpToChat: () => void,
-// isLoading: boolean,
-// }) => (
-// <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', minHeight: 48, position: 'relative'}}>
-// <ClickableBox
-// style={{
-// ...globalStyles.flexBoxCenter,
-// flexGrow: 1,
-// flexShrink: 0,
-// paddingLeft: globalMargins.medium,
-// paddingRight: globalMargins.medium,
-// }}
-// onClick={onJumpToChat}
-// >
-// <Box
-// style={{
-// ...globalStyles.flexBoxRow,
-// alignItems: 'center',
-// backgroundColor: globalColors.lightGrey,
-// borderRadius: 19,
-// flexShrink: 0,
-// height: globalMargins.medium,
-// justifyContent: 'center',
-// width: '100%',
-// }}
-// >
-// <Icon type="iconfont-search" style={{color: globalColors.black_20, fontSize: 12}} />
-// <Text type="Body" style={{color: globalColors.black_20, marginLeft: globalMargins.tiny}}>
-// Jump to chat
-// </Text>
-// </Box>
-// </ClickableBox>
-// <Icon
-// type="iconfont-add"
-// style={{color: globalColors.black_40, marginRight: 12, marginTop: 2}}
-// onClick={onNewChat}
-// />
-// {isLoading &&
-// <Box style={{bottom: 0, left: 0, position: 'absolute', right: 0}}>
-// <LoadingLine />
-// </Box>}
-// </Box>
-// )
+  _setFilter = filter => this._setState(this.state.inEditingMode, filter)
+
+  _enterEditingMode = () => {
+    this._setState(true, '')
+  }
+
+  _leaveEditingMode = () => {
+    // TODO: Set filter to '', but only after the chat is selected.
+    this._setState(false, this.state.filter)
+  }
+
+  render() {
+    let children
+    if (this.state.inEditingMode) {
+      children = [
+        <Icon
+          key="0"
+          type="iconfont-search"
+          style={{color: globalColors.grey, marginLeft: 9, marginRight: 9}}
+        />,
+        <Input
+          autoFocus={true}
+          key="1"
+          small={true}
+          value={this.state.filter}
+          hintText="Jump to..."
+          onChangeText={(filter: string) => this._setFilter(filter)}
+          onBlur={() => this._leaveEditingMode()}
+        />,
+      ]
+    } else {
+      children = (
+        <Box style={styleFilterContainer} onClick={() => this._enterEditingMode()}>
```

Comment:
we should avoid using closures as props in general. especially if the component updates a bunch (which this will). also this._enterEditingMode is already an arrow function so you can just do onClick={this._enterEditingMod}